runtime/lua/vim/lsp.lua

```diff
@@ -543,6 +544,12 @@ function lsp.start_client(config)
     env = config.cmd_env;
   })
 
+  if (config.languageId == nil) then
+    function config.languageId(filetype)
+      return filetype
+    end
+  end
+
```

Comment:
I'd just remove this part and above, where you do: 

```
local languageId = client.config.languageId(filetype)
```

I'd say something like

```
local languageId = client.config.language_id and client.config.language_id(filetype) or filetype
```